### PR TITLE
PT-1121 display parameter in daily view 

### PR DESCRIPTION
**Pull request by franck-fourel**

### CODE REVIEW

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import React, { PropTypes, PureComponent } from 'react';
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js:18]:
```suggestion
import React from 'react';
import PropTypes from 'prop-types';
```
let's be "propre" directly!

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import { formatLocalizedFromUTC, getHourMinuteFormat } from '../../../utils/datetime';
+
+import Tooltip from '../../common/tooltips/Tooltip';
+import colors from '../../../styles/colors.css';
+import styles from './ParameterTooltip.css';
+import i18next from 'i18next';
+
+const t = i18next.t.bind(i18next);
+
+class ParameterTooltip extends PureComponent {
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js:29]:
```suggestion
class ParameterTooltip extends React.Component {
```
This is not a PureComponent, let's use the standard Component

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  position: PropTypes.shape({
+    top: PropTypes.number.isRequired,
+    left: PropTypes.number.isRequired,
+  }).isRequired,
+  offset: PropTypes.shape({
+    top: PropTypes.number.isRequired,
+    left: PropTypes.number,
+    horizontal: PropTypes.number,
+  }),
+  title: PropTypes.node,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js:91]: Make the `title` a required property, or set a default value below

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  tail: PropTypes.bool.isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  tail: PropTypes.bool,
```
Not required, there is a default value.

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  side: PropTypes.oneOf(['top', 'right', 'bottom', 'left']).isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  side: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
```
Not required, there is a default value.

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  tailColor: PropTypes.string.isRequired,
+  tailWidth: PropTypes.number.isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  tailWidth: PropTypes.number,
```
Not required, there is a default value.

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  tailHeight: PropTypes.number.isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  tailHeight: PropTypes.number,
```
Not required, there is a default value.

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  }).isRequired,
+  offset: PropTypes.shape({
+    top: PropTypes.number.isRequired,
+    left: PropTypes.number,
+    horizontal: PropTypes.number,
+  }),
+  title: PropTypes.node,
+  tail: PropTypes.bool.isRequired,
+  side: PropTypes.oneOf(['top', 'right', 'bottom', 'left']).isRequired,
+  tailColor: PropTypes.string.isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  tailColor: PropTypes.string,
```
Not required, there is a default value.

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+  tailWidth: PropTypes.number.isRequired,
+  tailHeight: PropTypes.number.isRequired,
+  backgroundColor: PropTypes.string,
+  borderColor: PropTypes.string.isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  borderColor: PropTypes.string,
```
Not required, there is a default value.

File: src/components/daily/parametertooltip/ParameterTooltip.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+  borderWidth: PropTypes.number.isRequired,
```
Comment (yvon-dblg) [inline src/components/daily/parametertooltip/ParameterTooltip.js]:
```suggestion
  borderWidth: PropTypes.number,
```
Not required, there is a default value.

File: src/components/daily/physicaltooltip/PhysicalTooltip.js [hunk 11]
```diff
@@ -101,7 +101,7 @@ PhysicalTooltip.propTypes = {
     left: PropTypes.number,
     horizontal: PropTypes.number,
   }),
-  titls: PropTypes.node,
+  title: PropTypes.node,
```
Comment (yvon-dblg) [inline src/components/daily/physicaltooltip/PhysicalTooltip.js:104]: Missing `isRequired` or a default value.

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
With a few React fix, we're good to go :)

File: src/components/daily/physicaltooltip/PhysicalTooltip.js (hunk 11 above)
Comment (franck-fourel) [reply src/components/daily/physicaltooltip/PhysicalTooltip.js:104]: it's an optional property. 
Why would I need to set a default value ?

File: src/components/daily/parametertooltip/ParameterTooltip.js (hunk 1 above)
Comment (franck-fourel) [reply src/components/daily/parametertooltip/ParameterTooltip.js:18]: If I accept this one, I create an exception compared to other components. 
Lets' do that in a separate PR where we can address all components. 
https://diabeloop.atlassian.net/browse/PT-1146

File: src/components/daily/parametertooltip/ParameterTooltip.js (hunk 2 above)
Comment (franck-fourel) [reply src/components/daily/parametertooltip/ParameterTooltip.js:29]: same as above https://diabeloop.atlassian.net/browse/PT-1146

File: src/components/daily/physicaltooltip/PhysicalTooltip.js (hunk 11 above)
Comment (yvon-dblg) [reply src/components/daily/physicaltooltip/PhysicalTooltip.js:104]: That's how PropType works and have been think.
You can set the default value to null for example.

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**